includes/payments/functions.php

```diff
@@ -250,7 +250,7 @@ function give_create_payment( $payment_data ) {
 		'currency'        => give_get_currency( $form_id, $payment_data ),
 		'user_info'       => $payment_data['user_info'],
 		'status'          => 'pending',
-		'gateway'         => 'paypal',
+		'gateway'         => $payment_data['gateway'],
```

Comment:
@JasonTheAdams Currently recommend  way to create donation in GiveWP is  `give_insert_payment` function. I find that we are using `give_create_payment` only in `give_process_paypal_payment` which is PayPal related function, so hardcode value in this function is safe here but I recommend to move this function to `includes/gateways/paypal/paypal-standard.php` or deprecate it because this function confuses new developer.

Reply:
Sounds like we should deprecate this function, then. Is it possible to rewrite this function to internally using `give_insert_payment` and still retain the same signature?

Should @stochris switch to using `give_insert_payment` in the meantime?

Reply:
It's a bit confusing, since `includes/gateways/paypal/paypal-standard.php` is the only ( simple ) example of handling an async flow payment, I used the example there.

Of course, I'll use `give_insert_payment`, but I think the `paypal-standard.php` should be update to use the new method as well 😄  If I get around to it, I'd like to give it a shot ( hope I don't break anything 😓  )

About this PR, I'm open to either closing it, or merging it.

Reply:
@JasonTheAdams I am open to deprecate this function and update internal logic to use `give_insert_payment`.

cc: @stochris